permission_handler/android/src/main/java/com/baseflow/permissionhandler/PermissionUtils.java

```diff
@@ -279,20 +274,7 @@ static boolean isNeverAskAgainSelected(final Activity activity, final String nam
 
   @RequiresApi(api = Build.VERSION_CODES.M)
   static boolean neverAskAgainSelected(final Activity activity, final String permission) {
-    final boolean hasRequestedPermissionBefore = getRequestedPermissionBefore(activity, permission);
     final boolean shouldShowRequestPermissionRationale = ActivityCompat.shouldShowRequestPermissionRationale(activity, permission);
-    return hasRequestedPermissionBefore && !shouldShowRequestPermissionRationale;
-  }
-
-  static void setRequestedPermission(final Context context, final String permission) {
-    SharedPreferences genPrefs = context.getSharedPreferences("GENERIC_PREFERENCES", Context.MODE_PRIVATE);
-    SharedPreferences.Editor editor = genPrefs.edit();
-    editor.putBoolean(permission, true);
-    editor.apply();
-  }
-
-  static boolean getRequestedPermissionBefore(final Context context, final String permission) {
-    SharedPreferences genPrefs = context.getSharedPreferences("GENERIC_PREFERENCES", Context.MODE_PRIVATE);
-    return genPrefs.getBoolean(permission, false);
+    return !shouldShowRequestPermissionRationale;
   }
```

Comment:
Lets merge this method together with the `isNeverAskAgainSelected` method and clean up code a little bit:

```java
    @RequiresApi(api = Build.VERSION_CODES.M)
    static boolean isNeverAskAgainSelected(final Activity activity, final String name) {
        if (activity == null) {
            return false;
        }
        
        final boolean shouldShowRequestPermissionRationale = ActivityCompat.shouldShowRequestPermissionRationale(activity, permission);
        return !shouldShowRequestPermissionRationale; 
    }
```